### ISSUE TITLE
A way to pass data to commands

### ISSUE THREAD

**Issue by daveaglick** [spectresystems/spectre.cli#73]
What I want to do is register multiple commands using the same command type but pass different data to it when defining the commands. In other words, I want to be able to create a "dynamic" command at run time using the same base command.

The easiest way I can think to do this is add something like an `object Data { get; }` to the `CommandContext` and pass it through.

I imagine this might look something like:
```
public class FooCommandSettings : CommandSettings
{
    public string Bar { get; set; }
}

public class FooCommand : AsyncCommand<FooCommandSettings>
{
    public sealed override async Task<int> ExecuteAsync(
        CommandContext context, FooCommandSettings settings)
    {
        Console.WriteLine(context.Data + settings.Bar);
    }
}

public class Program
{
    public static int Main(string[] args)
    {
        var app = new CommandApp();

        app.Configure(config =>
        {
            // Set CommandContext.Data to "FIZZ"
            config.AddCommand<FooCommand >("fizz", "FIZZ");
            // Set CommandContext.Data to "FUZZ"
            config.AddCommand<FooCommand >("fuzz", "FUZZ");
        });

        return app.Run(args);
    }
}
```

Thoughts? Would you accept a PR?

**Comment by patriksvensson** [comment]
@daveaglick Sounds like a good idea to me! A PR is most welcome 👍

**Comment by daveaglick** [comment]
Resolved in #74
